websites/A/AniList/presence.ts

```diff
@@ -1,69 +1,145 @@
-		presenceData.smallImageText = "Searching";
-	} else if (pathname.startsWith("/anime")) {
-		presenceData.details = "Viewing an anime";
-		presenceData.state = document
-			.querySelector("div.content > h1")
-			.textContent.trim();
-	} else if (pathname.startsWith("/manga")) {
-		presenceData.details = "Viewing a manga";
-		presenceData.state = document
-			.querySelector("div.content > h1")
-			.textContent.trim();
-	} else if (pathname.startsWith("/forum")) {
-		if (pathname.split("/").length > 3) {
-			presenceData.details = "Reading a forum post";
-			presenceData.state = `'${document
-				.querySelector("h1.title")
-				.textContent.trim()}'`;
+		pathnameArray = document.location.pathname.split("/"),
+		page = pathnameArray[1],
+		showCover = await presence.getSetting<boolean>("cover");
+	strings = await getStrings();
+
+	switch (page) {
+		case "":
+			presenceData.details = strings.browsing;
+			presenceData.state = "Home";
+			break;
+		case "user": {
+			if (showCover) {
+				presenceData.largeImageKey = document
+					.querySelector(".avatar")
+					.getAttribute("src");
+				presenceData.smallImageKey = "anilist_lg";
+			}
+			switch (pathnameArray[3]) {
+				case "mangalist":
+					presenceData.details = "Viewing manga list";
+					break;
+				case "animelist":
+					presenceData.details = "Viewing anime list";
+					break;
+				case "favorites":
+					presenceData.details = "Viewing favorites";
+					break;
+				case "stats":
+					presenceData.details = "Viewing stats";
+					break;
+				case "social":
+					presenceData.details = "Viewing social page";
+					break;
+				case "reviews":
+					presenceData.details = "Viewing reviews";
+					break;
+				default:
+					presenceData.details = "Viewing profile";
+			}
+			presenceData.state = `from ${pathnameArray[2]}`;
+			presenceData.buttons = [
+				{
+					label: "View user's page",
+					url: document.location.href.replace(pathnameArray[3], ""),
+				},
+			];
+			break;
+		}
+		case "search":
+			presenceData.details = "Searching";
+			presenceData.smallImageKey = "search";
+			presenceData.smallImageText = "Searching";
+			break;
+		case "anime":
+		case "manga":
+			presenceData.details = `Viewing ${page === "anime" ? "an" : "a"} ${page}`;
+			presenceData.state = document
+				.querySelector("div.content > h1")
+				.textContent.trim();
+			if (showCover) {
+				presenceData.largeImageKey = document
+					.querySelector(".cover")
+					.getAttribute("src");
+			}
+			presenceData.buttons = [
+				{ label: `View ${page}`, url: document.location.href },
+			];
+			break;
+		case "character":
+		case "staff":
+			presenceData.details = `Viewing a ${page}`;
+			presenceData.state = document.querySelector(".name").textContent.trim();
+			if (showCover) {
+				presenceData.largeImageKey = document
+					.querySelector(".image")
+					.getAttribute("src");
+			}
+			presenceData.buttons = [
+				{ label: `View ${page}`, url: document.location.href },
```

Comment:
Dynamic label here too should be changed

Reply:
Okay, I was doubting about these too because of your previous review.

Reply:
I'm confused what do you mean? I think it is best not to have dynamic labels in buttons due to Discord's inconsitency and limitations

Reply:
Oh yes yes ! Sorry, I meant, when I see your review about the first button, I was thinking about these that were dynamic too, but I was not sure if the change was mandatory for them too.
By the way, resolved in 87074264d4a87d93a56008768b5a9a220eb9b3b7